src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -256,43 +256,57 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
+    if (this.search.match(/['"][^'"]+['"]/)) {
+      this.search = this.search.replace(/['"][^'"]+['"]/g, (match: string) => {
+        return match.replace(/(['"])([^'"]+)(['"])/, '$2').replace(' ', '+');
+      });
+    }
     // update the rows
-    this.rows = this.subSearch(this.data, this.search.toLowerCase().split(/[, ]/), columns);
+    this.rows = this.subSearch(this.data, this.search.toLowerCase().split(/[ ]/), columns);
     // Whenever the filter changes, always go back to the first page
     this.table.offset = 0;
   }
 
   subSearch (data: any[], currentSearch: string[], columns: CdTableColumn[]) {
-    let tempColumns: CdTableColumn[];
     if (currentSearch.length === 0 || data.length === 0) {
       return data;
     }
-    const searchWords: string[] = currentSearch.pop().split(':');
-    if (searchWords.length === 2) {
-      tempColumns = [...columns];
-      columns = columns.filter((c) => c.name.toLowerCase().indexOf(searchWords[0]) !== -1);
+    const searchTerms: string[] = currentSearch.pop().split(':');
+    const columnsClone = [...columns];
+    const dataClone = [...data];
+    const getSearchTerm = (s: string) => s.replace('+', ' ');
+    const filterColumns = (columnName: string) =>
+      columnsClone.filter((c) => c.name.toLowerCase().indexOf(columnName) !== -1);
+    if (searchTerms.length === 2) {
+      columns = filterColumns(getSearchTerm(searchTerms[0]));
     }
-    const searchWord: string = _.last(searchWords);
-    if (searchWord.length > 0) {
-      data = data.filter(d => {
-        return columns.filter(c => {
-          let cellValue: any = _.get(d, c.prop);
-          if (_.isUndefined(cellValue)) {
-            return;
-          }
-          if (_.isArray(cellValue)) {
-            cellValue = cellValue.join('');
-          } else if (_.isNumber(cellValue)) {
-            cellValue = cellValue.toString();
-          }
-          return cellValue.toLowerCase().indexOf(searchWord) !== -1;
-        }).length > 0;
-      });
+    const searchTerm: string = getSearchTerm(_.last(searchTerms));
+    data = this.basicDataSearch(searchTerm, data, columns);
+    // Checks if user searches for column but he is still typing
+    if (data.length === 0 && searchTerms.length === 1 && filterColumns(searchTerm).length > 0) {
+      data = dataClone;
     }
-    if (_.isArray(tempColumns)) {
-      columns = tempColumns;
+    return this.subSearch(data, currentSearch, columnsClone);
+  }
+
+  basicDataSearch(searchTerm: string, data: any[], columns: CdTableColumn[]) {
+    if (searchTerm.length === 0) {
+      return data;
     }
-    return this.subSearch(data, currentSearch, columns);
+    return data.filter(d => {
+      return columns.filter(c => {
+        let cellValue: any = _.get(d, c.prop);
+        if (_.isUndefined(cellValue)) {
+          return;
+        }
+        if (_.isArray(cellValue)) {
+          cellValue = cellValue.join('');
```

Comment:
We should use the same char that we normally use to display the tables.

I will list some tests I made on the OSD page where all rows have "up, in" as the Status.

- `up, in` -> Didn't return anything. **Maybe** this should return all rows, but I'm ok with this.
- `up in` -> Returned all rows. **Correct**.
- `"up, in"` ->  Didn't return anything. Should return all rows, because I typed exactly what I see in the rows.
- `"up in"` -> Didn't return anything. **Correct**.
- `status:up status:in` -> Returned all rows. **Correct**.
- `pi` -> Return all rows. Should have returned empty.

Reply:
Commas are ignored now.
I couldn't reproduce the `pi` example it will return nothing for me.
Now all other examples will show all up and in rows.
It's OK if you miss the comma in quotes as you are not searching any string, you are searching an array. Of course ignoring the comma will loose the way to find strings with a comma in name but those strings a pretty rare, comma is mostly used as separator for items.